src/server.ts

```diff
@@ -187,11 +209,22 @@ export async function startServer<RawConfiguration, Configuration, State>(
       }>
     ) => {
       request.log.debug({ requestBody: request.body }, "Query Request");
-      const queryResponse = await connector.query(
-        configuration,
-        state,
-        request.body
+
+      const queryResponse = await tracer.startActiveSpan(
+        "handleQuery",
+        async (span) => {
+          const queryResponse = await connector.query(
+            configuration,
+            state,
+            request.body
+          );
+
+          span.end();
+
+          return queryResponse;
+        }
```

Comment:
With this tracing pattern, what happens if `connector.query` throws and `span.end` does not get called?